src/main/java/com/minecolonies/client/gui/WindowHutFarmer.java

```diff
@@ -201,37 +188,122 @@ public String getBuildingName()
+
+                rowPane.findPaneOfTypeByID(TAG_WORKER, Label.class).setLabelText(owner);
+                rowPane.findPaneOfTypeByID(TAG_DISTANCE, Label.class).setLabelText(distance  + "m");
+
+                rowPane.findPaneOfTypeByID(TAG_DIRECTION, Label.class).setLabelText(direction);
+                rowPane.findPaneOfTypeByID(TAG_HIDDEN_ID, Label.class).setLabelText(Integer.toString(index));
+
+                if(!building.assignFieldManually())
+                {
+                    rowPane.findPaneOfTypeByID(TAG_BUTTON_ASSIGN, Button.class).disable();
+                }
+
+                if (field.isTaken())
+                {
+                    rowPane.findPaneOfTypeByID(TAG_BUTTON_ASSIGN, Button.class).setLabel(RED_X);
+                }
+                else
+                {
+                    rowPane.findPaneOfTypeByID(TAG_BUTTON_ASSIGN, Button.class).setLabel(APPROVE);
+                    if(building.getBuildingLevel() <= building.getAmountOfFields())
+                    {
+                        rowPane.findPaneOfTypeByID(TAG_BUTTON_ASSIGN, Button.class).disable();
+                    }
+                }
+
+                if(field.getItem() != null)
+                {
+                    rowPane.findPaneOfTypeByID(TAG_ICON, ItemIcon.class).setItem(new ItemStack(field.getItem(),1));
+                }
+            }
+        });
     }
 
     /**
-     * Updates the labels on the buttons
+     * Calculates the direction the field is from the building.
+     * @param building the building
+     * @param field the field.
+     * @return a string describing the direction.
      */
-    private void updateButtonLabels()
+    private String calcDirection(BlockPos building, BlockPos field)
     {
-        try
+        String dist = "";
+
+        if(field.getZ() > building.getZ()+1)
+        {
+            dist = LanguageHandler.format("com.minecolonies.gui.workerHuts.farmerHut.South");
+        }
+        else if(field.getZ() < building.getZ()-1)
         {
-            findPaneOfTypeByID(BUTTON_WHEAT, ButtonVanilla.class).setLabel(Integer.toString(building.wheat));
-            findPaneOfTypeByID(BUTTON_POTATO, ButtonVanilla.class).setLabel(Integer.toString(building.potato));
-            findPaneOfTypeByID(BUTTON_CARROT, ButtonVanilla.class).setLabel(Integer.toString(building.carrot));
-            findPaneOfTypeByID(BUTTON_MELON, ButtonVanilla.class).setLabel(Integer.toString(building.melon));
-            findPaneOfTypeByID(BUTTON_PUMPKIN, ButtonVanilla.class).setLabel(Integer.toString(building.pumpkin));
+            dist = LanguageHandler.format("com.minecolonies.gui.workerHuts.farmerHut.North");
+        }
 
+        if(field.getX() > building.getX()+1)
+        {
+            dist += LanguageHandler.format("com.minecolonies.gui.workerHuts.farmerHut.East");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Prefer StringBuffer over += for concatenating strings [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AUseStringBufferForStringAppends)